flink-docs/src/main/java/org/apache/flink/docs/configuration/ConfigOptionsDocGenerator.java

```diff
@@ -260,10 +262,26 @@ private static String toHtmlTable(final List<OptionWithMetaInfo> options) {
 	private static String toHtmlString(final OptionWithMetaInfo optionWithMetaInfo) {
 		ConfigOption<?> option = optionWithMetaInfo.option;
 		String defaultValue = stringifyDefault(optionWithMetaInfo);
+		Documentation.TableOption tableOption = optionWithMetaInfo.field.getAnnotation(Documentation.TableOption.class);
+		StringBuilder sb = new StringBuilder();
+		if (tableOption != null) {
+			Documentation.ExecMode execMode = tableOption.execMode();
+			if (Documentation.ExecMode.BATCH.equals(execMode) || Documentation.ExecMode.STREAMING.equals(execMode)) {
+				sb.append("<br> <span class=\"label label-primary\">")
+						.append(execMode.toString())
+						.append("</span>");
+			} else if (Documentation.ExecMode.BOTH.equals(execMode)) {
```

Comment:
unnecessarily complex if statement.
Either use
```
if (BOTH.equals(execMode) {
...
} else {
...
}
```
or introduce a switch statement with a fast-failing default branch.

Reply:
Fixed it.